tests/integration/test_wazuh_db/test_sync_agent_groups_get.py

```diff
@@ -0,0 +1,135 @@
+tags:
+    - wazuh_db
+'''
+import os
+import time
+import pytest
+import yaml
+import json
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.wazuh_db import (query_wdb, insert_agent_into_group, clean_agents_from_db,
+                                    clean_groups_from_db, clean_belongs)
+
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Configurations
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+messages_file = os.path.join(os.path.join(test_data_path, 'global'), 'sync_agent_groups_get.yaml')
+module_tests = []
+with open(messages_file) as f:
+    module_tests.append((yaml.safe_load(f), messages_file.split('_')[0]))
+
+log_monitor_paths = []
+wdb_path = os.path.join(os.path.join(WAZUH_PATH, 'queue', 'db', 'wdb'))
+receiver_sockets_params = [(wdb_path, 'AF_UNIX', 'TCP')]
+monitored_sockets_params = [('wazuh-db', None, True)]
+receiver_sockets = None  # Set in the fixtures
+agents = ['agent1', 'agent2']
+
+
+# Fixtures
+@pytest.fixture(scope='function')
+def pre_insert_agents():
+    insert_agent_into_group(2)
+
+    yield
+    clean_agents_from_db()
+    clean_groups_from_db()
+    clean_belongs()
+
+
+# Tests
+@pytest.mark.parametrize('test_case',
+                         [case['test_case'] for module_data in module_tests for case in module_data[0]],
+                         ids=[f"{module_name}: {case['name']}"
+                              for module_data, module_name in module_tests
+                              for case in module_data]
+                         )
+def test_sync_agent_groups(configure_sockets_environment, connect_to_sockets_module, test_case, pre_insert_agents):
+    '''
+    description: Check that commands about sync_aget_groups_get works properly.
+    wazuh_min_version: 4.4.0
+    parameters:
+        - configure_sockets_environment:
+            type: fixture
+            brief: Configure environment for sockets and MITM.
+        - connect_to_sockets_module:
+            type: fixture
+            brief: Module scope version of 'connect_to_sockets' fixture.
+        - test_case:
+            type: fixture
+            brief: List of test_case stages (dicts with input, output and agent_id and expected_groups keys).
+        - pre_insert_agents:
+            type: fixture
+            brief: fixture in charge of insert agents and groups into DB.
+    assertions:
+        - Verify that the socket response matches the expected output.
+    input_description:
+        - Test cases are defined in the sync_agent_groups_get.yaml file.
+    expected_output:
+        - an array with all the agents that match with the search criteria
+    tags:
+        - wazuh_db
+        - wdb_socket
+    '''
+    case_data = test_case[0]
+    output = case_data["output"]
+
+    if 'pre_input' in case_data:
+        for command in case_data['pre_input']:
+            query_wdb(command)
+            results = query_wdb(command)
```

Comment:
Before doing this, add a comment like "check if it requires any special configuration".